FirebaseAuth/Sources/Utilities/FIRAuthRecaptchaVerifier.m

```diff
@@ -0,0 +1,252 @@
+
+    providerToStringMap = @{@(FIRAuthRecaptchaProviderPassword) : @"EMAIL_PASSWORD_PROVIDER"};
+
+    actionToStringMap = @{
+      @(FIRAuthRecaptchaActionSignInWithPassword) : @"signInWithPassword",
+      @(FIRAuthRecaptchaActionGetOobCode) : @"getOobCode",
+      @(FIRAuthRecaptchaActionSignUpPassword) : @"signUpPassword"
+    };
+  });
+  if (sharedRecaptchaVerifier.auth != auth) {
+    sharedRecaptchaVerifier.agentConfig = nil;
+    sharedRecaptchaVerifier.tenantConfigs = nil;
+    sharedRecaptchaVerifier.auth = auth;
+  }
+  return sharedRecaptchaVerifier;
+}
+
+- (NSString *)siteKey {
+  if ([FIRAuth auth].tenantID == nil) {
+    return self->_agentConfig.siteKey;
+  } else {
+    FIRAuthRecaptchaConfig *config = self->_tenantConfigs[[FIRAuth auth].tenantID];
+    if (config) {
+      return config.siteKey;
+    } else {
+      return nil;
+    }
+  }
+}
+
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+- (BOOL)enablementStatusForProvider:(FIRAuthRecaptchaProvider)provider {
+  if ([FIRAuth auth].tenantID == nil) {
+    return [self->_agentConfig.enablementStatus[providerToStringMap[@(provider)]] boolValue];
+  } else {
+    return [self->_tenantConfigs[[FIRAuth auth].tenantID]
+                .enablementStatus[providerToStringMap[@(provider)]] boolValue];
+  }
+}
+
+- (void)verifyForceRefresh:(BOOL)forceRefresh
+                    action:(FIRAuthRecaptchaAction)action
+                completion:(nullable FIRAuthRecaptchaTokenCallback)completion {
+  [self
+      retrieveRecaptchaConfigForceRefresh:forceRefresh
+                               completion:^(NSError *_Nullable error) {
+                                 if (error) {
+                                   completion(nil, error);
+                                 }
+                                 if (!self.recaptchaClient) {
+                                   NSString *siteKey = [self siteKey];
+                                   Class RecaptchaClass = NSClassFromString(@"Recaptcha");
+                                   if (RecaptchaClass) {
+                                     SEL selectorWithTimeout = NSSelectorFromString(
+                                         @"getClientWithSiteKey:withTimeout:completion:");
+                                     if ([RecaptchaClass respondsToSelector:selectorWithTimeout]) {
+                                       void (*funcWithTimeout)(
+                                           id, SEL, NSString *, double,
+                                           void (^)(
+                                               id<RCARecaptchaClientProtocol> _Nullable recaptchaClient,
+                                               NSError *_Nullable error)) = (void *)[RecaptchaClass
+                                           methodForSelector:selectorWithTimeout];
+                                       funcWithTimeout(
+                                           RecaptchaClass, selectorWithTimeout, siteKey, 10000.0,
```

Comment:
Why 10000?

Reply:
10000 (10s) is the default timeout value. While it seems we don't need to customize this value, so removed this parameter to just use the default value.